PyTango mandatory device property with default value

https://github.com/tango-controls/pytango/blob/d467c1be41ab0ff888a5a6cd7041a9780ff40468/tango/server.py#L1214

Does not that mean that there is no way to define a mandatory property with a default value, does it?

@Ingvord OK, I've made a PR with a new message, combining some of your's and some of mine.

I don't want to change the behaviour, so I'm not ignoring the default - still raising the exception, so the developer can fix the error.

The name of the field is not available, because the error happens before completing the assignment to the field.  However, the location of the problem is obvious from the stack trace.  See the example of `my_bad_property` below:

```
tests/test_test_context.py:None (tests/test_test_context.py)
test_test_context.py:18: in <module>
    class Device1(Device):
test_test_context.py:19: in Device1
    my_bad_property = device_property(dtype=int, mandatory=True, default_value=123)
../tango/server.py:1218: in __init__
    raise ValueError(msg)
E   ValueError: Invalid arguments: 'mandatory' is True, so 'default_value' must be None. A mandatory device property value must be defined in the Tango Database so it cannot have a default.

```

--- a/tango/server.py
+++ b/tango/server.py
@@ -1212,9 +1212,10 @@ class device_property(_BaseProperty):
                                               update_db)
         self.mandatory = mandatory
         if mandatory and default_value is not None:
-            msg = "device_property arguments mandatory " \
-                  "and default_value are incompatible"
-            raise Exception(msg)
+            msg = ("Invalid arguments: 'mandatory' is True, so 'default_value' must be None. "
+                   "A mandatory device property value must be defined in the Tango Database "
+                   "so it cannot have a default.")
+            raise ValueError(msg)
 
 
 class class_property(_BaseProperty):

